src/sentry/api/endpoints/organization_invite_request_details.py

```diff
@@ -1,14 +1,202 @@
+                Q(invite_status=InviteStatus.REQUESTED_TO_BE_INVITED.value)
+                | Q(invite_status=InviteStatus.REQUESTED_TO_JOIN.value),
+                organization=organization,
+                user__isnull=True,
+                id=member_id,
+            )
+        except ValueError:
+            raise OrganizationMember.DoesNotExist()
+
+    def get(self, request, organization, member_id):
+        try:
+            member = self._get_member(organization, member_id)
+        except OrganizationMember.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        return Response(
+            serialize(member, serializer=OrganizationMemberWithTeamsSerializer()),
+            status=status.HTTP_200_OK,
+        )
+
+    def put(self, request, organization, member_id):
+        """
+        Update an invite request to Organization
+        ````````````````````````````````````````
+
+        Update and/or approve an invite request to an organization.
+
+        :pparam string organization_slug: the slug of the organization the member will belong to
+        :param string member_id: the member ID
+        :param boolean approve: allows the member to be invited
+        :param string role: the suggested role of the new member
+        :param array teams: the suggested slugs of the teams the member should belong to.
+
+        :auth: required
+        """
+
+        try:
+            member = self._get_member(organization, member_id)
+        except OrganizationMember.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        if (
+            member.inviter != request.user
+            and not request.access.has_scope("member:admin")
+            and not request.access.has_scope("member:write")
+        ):
+            return Response(
+                {"detail": ERR_INSUFFICIENT_ROLE_UPDATE}, status=status.HTTP_403_FORBIDDEN
+            )
+
+        serializer = OrganizationMemberSerializer(
+            data=request.data,
+            context={"organization": organization, "allowed_roles": roles.get_all()},
+            partial=True,
+        )
+
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        result = serializer.validated_data
+
+        if result.get("role"):
+            member.update(role=result["role"])
+
+        if "teams" in result:
+            save_team_assignments(member, result["teams"])
+
+        if "approve" in request.data:
+            serializer = InviteRequestSerializer(
+                data=request.data,
+                context={"request": request, "organization": organization, "member": member},
+            )
+
+            if not serializer.is_valid():
+                return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+            result = serializer.validated_data
+
+            if result.get("approve"):
+                if not member.invite_approved:
```

Comment:
We can condense this I think `if result.get("approve") and not member.invite_approved:`